worlds/osrs/__init__.py

```diff
@@ -0,0 +1,616 @@
+        self.region_rows_by_name = {}
+        self.resource_rows_by_name = {}
+        self.item_rows_by_name = {}
+
+        self.starting_area_item = ""
+        self.allow_brutal_grinds = False
+
+        self.location_categories = {}
+        self.locations_by_category = {}
+
+    def generate_early(self) -> None:
+        self.location_categories = {location_row.category for location_row in load_location_csv()}
+        self.locations_by_category = {category:
+                                          [location_row for location_row in self.location_rows if
+                                           location_row.category == category]
+                                      for category in self.location_categories}
+
+        self.location_rows_by_name = {loc_row.name: loc_row for loc_row in self.location_rows}
+        self.region_rows_by_name = {reg_row.name: reg_row for reg_row in self.region_rows}
+        self.resource_rows_by_name = {rec_row.name: rec_row for rec_row in self.resource_rows}
+        self.item_rows_by_name = {it_row.name: it_row for it_row in self.item_rows}
+
+        rnd = self.multiworld.per_slot_randoms[self.player]
+        starting_area = self.multiworld.starting_area[self.player]
+        self.allow_brutal_grinds = self.multiworld.brutal_grinds[self.player]
+
+        if starting_area.value == StartingArea.option_any_bank:
+            random_bank = rnd.randint(0, len(starting_area_dict) - 1)
+            self.starting_area_item = starting_area_dict[random_bank]
+        elif starting_area.value < StartingArea.option_chunksanity:
+            self.starting_area_item = starting_area_dict[starting_area.value]
+        else:
+            chunksanity_random = rnd.randint(0, len(chunksanity_starting_chunks) - 1)
+            self.starting_area_item = chunksanity_starting_chunks[chunksanity_random]
+
+        # Set Starting Chunk
+        self.multiworld.push_precollected(self.create_item(self.starting_area_item))
+
+    def create_regions(self) -> None:
+        """
+        called to place player's regions into the MultiWorld's regions list. If it's hard to separate, this can be done
+        during generate_early or basic as well.
+        """
+
+        # First, create the "Menu" region to start
+        menu_region = self.create_region("Menu")
+
+        for region_row in self.region_rows:
+            self.create_region(region_row.name)
+
+        for resource_row in self.resource_rows:
+            self.create_region(resource_row.name)
+
+        # Removes the word "Area: " from the item name to get the region it applies to.
+        # I figured tacking "Area: " at the beginning would make it _easier_ to tell apart. Turns out it made it worse
+        starting_area_region = self.starting_area_item[len("Area: "):]
+        starting_entrance = menu_region.create_exit(f"Start->{starting_area_region}")
+        starting_entrance.access_rule = lambda state: state.has(self.starting_area_item, self.player)
```

Comment:
Solely for the purposes of having the starting area item show in the "inventory" in Runelite. I figured if I had to have the item in there anyway I might as well use it in all the other places for consistency. It might be a bit over-designed for no benefit though.

Reply:
That's fair enough. You could probably still skip making the access rule, since it would be a small performance gain, but also it's whatever

Reply:
Ah, now I remember why. The Starting Location rule was set so that a spoiler log would show the starting area as sphere 0 to make the playthrough a bit easier to read. I can remove it if we need the speed boost, but I think fixing the skill checks described below would be a bigger impact, it might not be necessary after that.